### PR TITLE
feat: Create guild cache

### PR DESCRIPTION
**Pull request by ySnoopyDogy**

### CODE REVIEW

File: src/database/Databases.ts [hunk 1]
```diff
@@ -59,7 +66,7 @@ export default class MongoDatabase {
 
   private readonly giveRepository: GiveRepository;
 
-  constructor(public uri: string) {
+  constructor(public uri: string, wantRedisCache: boolean) {
```
Comment (almeidx) [inline src/database/Databases.ts:69]: `wantRedisCache` sounds odd. Perhaps name it `withRedis` or `withRedisCache` instead?

File: src/database/Databases.ts [hunk 2]
```diff
@@ -125,6 +120,18 @@ export default class MongoDatabase {
...
+    return new Promise((resolve, reject) => {
+      try {
+        this.redisClient = new Redis({ path: process.env.REDIS_PATH as string });
+        resolve();
+      } catch (err) {
+        console.log(`[REDIS] Error connecting to redis ${err}`);
+        reject(err);
+      }
+    });
+  }
```
Comment (almeidx) [inline src/database/Databases.ts:133]: Not sure why you opted for a promise in the first place

```suggestion
  createRedisConnection(): void {
    try {
      this.redisClient = new Redis({ path: process.env.REDIS_PATH as string });
    } catch (err) {
      console.log(`[REDIS] Error connecting to redis ${err}`);
      throw err;
    }
  }
```

File: src/database/repositories/CacheRepository.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    if (this.redisClient) {
+      const dataAtual = await this.redisClient.get(`guild:${guildID}`);
+      if (dataAtual) {
+        const jsonData = JSON.parse(dataAtual);
+        const index = jsonData.disabledCommands.indexOf(commandName);
+        jsonData.disabledCommands.splice(index, 1);
+        const afterChanges = JSON.stringify(jsonData);
+        await this.redisClient.set(`guild:${guildID}`, afterChanges);
+      }
+    }
```
Comment (almeidx) [inline src/database/repositories/CacheRepository.ts:40]: IMO this pattern is cleaner, and it uses one less scope. If you like it as well, you can replicate it on the rest of the methods in this file

```suggestion
    redisBlock: if (this.redisClient) {
      const cachedData = await this.redisClient.get(`guild:${guildID}`);
      if (!cachedData) break redisBlock;
      const jsonData = JSON.parse(cachedData);
      const index = jsonData.disabledCommands.indexOf(commandName);
      jsonData.disabledCommands.splice(index, 1);
      await this.redisClient.set(`guild:${guildID}`, JSON.stringify(jsonData));
    }
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/database/repositories/CacheRepository.ts (hunk 3 above)
Comment (ySnoopyDogy) [reply src/database/repositories/CacheRepository.ts:40]: I tried to use it, but it breakes 2 rules of my eslint, so i wont use it, but thanks for the suggestion, i didnt know the tabled statements, it seems to be very util

**Review by Tsugami — CHANGES_REQUESTED** [review]
I prefer to separate set and get from redis into a separate file, lots of things repeated like JSON.stringify always in set

**Pull request merged**